addons/mail/models/mail_thread.py

```diff
@@ -1304,7 +1304,9 @@ def _message_extract_payload(self, message, save_original=False):
         # Content-Type: multipart/related;
         #   boundary="_004_3f1e4da175f349248b8d43cdeb9866f1AMSPR06MB343eurprd06pro_";
         #   type="text/html"
-        if not message.is_multipart() or message.get('content-type', '').startswith("text/"):
+        main_type = message.get('content-type', '').split('/')[0]
+        binary_types = ('application', 'audio', 'image', 'video')
+        if (not message.is_multipart and main_type not in binary_types) or main_type == 'text':
```

Comment:
I suppose you meant to keep the parentheses after `message.is_multipart`?
That said, I can't find any use case where the first part of the `if` condition would make sense. Shouldn't we simplify this to just the `text/*` Content-Type case?

Reply:
Ouch, those parentheses...
Agreed! I switch to get_content_maintype() which always returns a sane value.